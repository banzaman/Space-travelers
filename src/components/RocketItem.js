import React from 'react';
import { useDispatch } from 'react-redux';
import PropTypes from 'prop-types';
import { reserveRocket } from '../redux/rockets/rocketsSlice';
import '../styles/rockets.css';

const RocketItem = ({
  id, name, Image, description, reserved,
}) => {
  const dispatch = useDispatch();
  if (!reserved) {
    return (
      <div className="rocketItemCont">
        <section className="rocketImgCont">
          <img src={Image[0]} alt="" />
        </section>
        <section id={id} className="rocket_details">
          <h2>{name }</h2>
          <p>
            {description}
          </p>
          <button className="reserveButton" type="button" value="Reserve Rocket" onClick={() => (dispatch(reserveRocket(id)))}>Reserve Rocket</button>
        </section>
      </div>
    );
  }
  return (
    <div className="rocketItemCont">
      <section className="rocketImgCont">
        <img src={Image[0]} alt="" />
      </section>
      <section id={id} className="rocket_details">
        <h2>{name }</h2>
        <p className="description">
          <span className="reservedTag">Reserved</span>
          {description}
        </p>
        <button className="cancelButton" type="button" value="Reserve Rocket" onClick={() => (dispatch(reserveRocket(id)))}>Cancel Reservation</button>
      </section>
    </div>
  );
};

RocketItem.propTypes = { id: PropTypes.string.isRequired };
RocketItem.propTypes = { name: PropTypes.string.isRequired };
RocketItem.propTypes = { Image: PropTypes.string.isRequired };
RocketItem.propTypes = { description: PropTypes.string.isRequired };
RocketItem.propTypes = { reserved: PropTypes.bool.isRequired };

export default RocketItem;
