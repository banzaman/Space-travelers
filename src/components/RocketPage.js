import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { getRocketItems } from '../redux/rockets/rocketsSlice';

const RocketPage = () => {
  const { isLoading, rocketItems } = useSelector((store) => store.rockets);
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(getRocketItems());
  }, [dispatch]);

  if (isLoading) {
    return (
      <div className="loading">
        <h1>Loading...</h1>
      </div>
    );
  }
  return (
    console.log(rocketItems)
  );
};

export default RocketPage;
