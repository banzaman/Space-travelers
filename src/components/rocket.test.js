import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/store';
import RocketItem from './RocketItem';
import RocketPage from './RocketPage';
import RocketTest from './rocketsReducer';

const rocket = {
  id: '2',
  name: 'Falcon 2',
  description: 'Good rocket!',
  flickr_images: 'hi',
  reserved: false,
};

jest.mock('axios');
describe('Rockets must pass the test', () => {
  test('added Rockets must return data', () => {
    expect(RocketTest.addedRockets()).toBeDefined();
  });
  test('added Rockets return value length must be 4', () => {
    expect(RocketTest.addedRockets()).toHaveLength(3);
  });
  test('added Rockets must return name t', () => {
    expect(RocketTest.addedRockets()[2].description).toBe('Amazing rocket!');
  });
});

describe('Rockets component', () => {
  it('renders Rockets component', () => {
    const rendering = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <RocketPage />
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(rendering).toMatchSnapshot();
  });
  it('renders Rocket component', () => {
    const itemMock = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <RocketItem
            id={rocket.id}
            name={rocket.name}
            description={rocket.description}
            Image={rocket.flickr_images}
            reserved={rocket.reserved}
          />
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(itemMock).toMatchSnapshot();
  });
});
