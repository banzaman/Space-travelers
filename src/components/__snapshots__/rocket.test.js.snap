// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rockets component renders Rocket component 1`] = `
<div
  className="rocketItemCont"
>
  <section
    className="rocketImgCont"
  >
    <img
      alt=""
      src="h"
    />
  </section>
  <section
    className="rocket_details"
    id="2"
  >
    <h2>
      Falcon 2
    </h2>
    <p>
      Good rocket!
    </p>
    <button
      className="reserveButton"
      data-testid="reserveButton"
      onClick={[Function]}
      type="button"
    >
      Reserve Rocket
    </button>
  </section>
</div>
`;

exports[`Rockets component renders Rockets component 1`] = `
<div
  className="loading"
>
  <h1>
    Loading...
  </h1>
</div>
`;
